hledger-lib/Hledger/Data/MarketPrice.hs

```diff
@@ -12,22 +12,35 @@ are thousands separated by comma, significant decimal places and so on.
 
 -}
 
-{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings, LambdaCase #-}
 
 module Hledger.Data.MarketPrice
 where
 import qualified Data.Text as T
--- import Test.HUnit
+import Test.HUnit
 
 import Hledger.Data.Amount
-import Hledger.Data.Types
 import Hledger.Data.Dates
+import Hledger.Data.Types
+-- import Hledger.Read.JournalReader
 import Hledger.Utils
+import Hledger.Utils.Parse
 
+-- | Get the string representation of an market price, based on its
+-- commodity's display settings.
 showMarketPrice :: MarketPrice -> String
 showMarketPrice mp = unwords
     [ "P"
     , showDate (mpdate mp)
     , T.unpack (mpcommodity mp)
     , (showAmount . setAmountPrecision maxprecision) (mpamount mp)
```

Comment:
```
zsh% hledger prices -f -
P 2016/07/01 VND $0.000045
P 2016-07-01 VND $0.000045
```
But with:

```
zsh% hledger prices -f -
commodity $
    format $1,000,000,000.00
P 2016/07/01 VND $0.000045
P 2016-07-01 VND $0.00
```

That's a bug.`commodity` directive shouldn't affect prices. This may result in truncating prices in output of `hledger print`